sdks/python/apache_beam/testing/test_utils.py

```diff
@@ -79,7 +79,7 @@ def compute_hash(content, hashing_alg=DEFAULT_HASHING_ALG):
   content.sort()
   m = hashlib.new(hashing_alg)
   for elem in content:
-    m.update(str(elem).encode('utf-8'))
+    m.update(elem)
```

Comment:
This fails on `None` values introduced in [this test](https://github.com/apache/beam/blob/8db6b63276f6e749f095b5817b22efb431e22e6b/sdks/python/apache_beam/io/gcp/tests/bigquery_matcher_test.py#L54).

However the documentation of the current function indicates that it expects a list of strings. I also can't think of a scenario where BigQuery returns a list of `None` values for rows from a query. I'm therefore wondering if this is a problem with the current function, or the test should be changed.

Previously, this function just casted `None` to `"None"` before calculating the hash.

Reply:
PTAL @tvalentyn @markflyhigh

Reply:
On Python 3, `str(None).encode('utf-8')` results in `'b"None"'`. Checking for `bytes` fixes this.
I also moved this conversion to the start of the method, since `sort()` otherwise fails with a `TypeError` on Python 3 because of unorderable types.